Задание 4 "Диаграммы активности и использования"

yeputons opened the pull request:
Это диаграммы для приложения O'Feed (https://github.com/yeputons/ofeed), которое предназначено для оффлайн-просмотра ленты новостей ВК. Должно уметь подгружать ленту "кусками" и автоматически подгружать фотографии и страницы по ссылкам из новостей, чтобы потом показывать их без интернета. На диаграмме активности показан процесс обновления ленты в некотором месте.

yeputons committed:
Add activity and use case diagrams for O'Feed

yeputons committed:
Add cover.txt

yeputons committed:
Update 04-activity and save it as PNG

yeputons committed:
Another update of 04-activity; add more actors and details

yeputons committed:
04-activity: fix cycle scheme

yurii-litvinov commented:
Диаграмма случаев использования получилась чем-то средним между диаграммой случаев использования и dataflow. Например, вконтакт реально **пользуется** функцией "скачать посты" Вашего приложения? И многочисленные зависимости между случаями использования внутри системы тоже намекают на желание показать, кто кого вызывает и что делает каждая компонента системы. Это всё не нужно. На этой диаграмме рисуются группы пользователей (роли) и что они могут делать с системой, причём тут отображаются их высокоуровневые "желания", типа "смотреть ленту новостей". Даже авторизация --- это не случай использования, потому что пользователи ставят приложение не для того, чтобы авторизоваться. Авторизация --- это часть сценария "просмотреть ленту", и должна рисоваться уже на диаграмме активностей. Диаграмма случаев использования вообще должна быть простой (есть эвристическое правило рисовать не больше семи случаев использования на диаграмме).

Диаграмма активностей мне нравится, а вот диаграмму случаев использования надо перерисовать.

yeputons committed:
Update 04-use-case according to review comments; save image as PNG now

yeputons commented:
@yurii-litvinov Попробовал исправить. Не уверен, правда, в типе связи между ВК/сторонними серверами и приложением. Кажется, что они точно должны быть (без ВК/сторонних серверов приложение не имеет смысла), но ведь это связь в некотором смысле односторонняя: приложению важно, а им всё равно.

yurii-litvinov commented:
Так а ВК реально пользуется системой или он источник данных? Если источник, то на этой диаграмме он не рисуется, пользователю же всё равно, откуда берутся данные, лишь бы это была его лента из ВК. Короче, это деталь реализации тут

yeputons committed:
Update 04-use-case: remove external services

They're used as data sources only and user do not care about the source of
their data. So, external services are not included in use case diagram.

yeputons commented:
@yurii-litvinov Спасибо за пояснение, снёс.
